airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/mssql/MssqlDebeziumStateUtil.java

```diff
@@ -0,0 +1,157 @@
+        Optional.empty());
+    final AirbyteSchemaHistoryStorage schemaHistoryStorage =
+        AirbyteSchemaHistoryStorage.initializeDBHistory(new SchemaHistory<>(Optional.empty(), false), false);
+    final LinkedBlockingQueue<ChangeEvent<String, String>> queue = new LinkedBlockingQueue<>();
+    final Instant engineStartTime = Instant.now();
+    try {
+      final DebeziumRecordPublisher publisher = new DebeziumRecordPublisher(properties,
+          database.getSourceConfig(),
+          catalog,
+          emptyOffsetManager,
+          Optional.of(schemaHistoryStorage),
+          DebeziumPropertiesManager.DebeziumConnectorType.RELATIONALDB);
+      publisher.start(queue);
+      while (!publisher.hasClosed()) {
+        final ChangeEvent<String, String> event = queue.poll(10, TimeUnit.SECONDS);
+        if (event != null) {
+          publisher.close();
+          break;
+        }
+        if (Duration.between(engineStartTime, Instant.now()).compareTo(Duration.ofMinutes(3)) > 0) {
+          LOGGER.error("No record is returned even after {} seconds of waiting, closing the engine", 180);
+          publisher.close();
+          throw new RuntimeException(
+              "Building schema history has timed out. Please consider increasing the debezium wait time in advanced options.");
+        }
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    final AirbyteFileOffsetBackingStore offsetManager = AirbyteFileOffsetBackingStore.initializeState(highWaterMark,
+        Optional.empty());
+
+    final Map<String, String> offset = offsetManager.read();
+    final SchemaHistory<String> schemaHistory = schemaHistoryStorage.read();
+
+    assert !offset.isEmpty();
+    assert Objects.nonNull(schemaHistory);
+    assert Objects.nonNull(schemaHistory.schema());
+
+    final JsonNode asJson = serialize(offset, schemaHistory);
+    LOGGER.info("Initial Debezium state constructed: {}", asJson);
+
+    if (asJson.get(MssqlCdcStateConstants.MSSQL_DB_HISTORY).asText().isBlank()) {
+      throw new RuntimeException("Schema history snapshot returned empty history.");
+    }
+    return asJson;
+
+  }
+
+  public static JsonNode serialize(final Map<String, String> offset, final SchemaHistory<String> dbHistory) {
+    final Map<String, Object> state = new HashMap<>();
+    state.put(MssqlCdcStateConstants.MSSQL_CDC_OFFSET, offset);
+    state.put(MssqlCdcStateConstants.MSSQL_DB_HISTORY, dbHistory.schema());
+    state.put(MssqlCdcStateConstants.IS_COMPRESSED, dbHistory.isCompressed());
+
+    return Jsons.jsonNode(state);
+  }
+
+  public static MssqlDebeziumStateAttributes getStateAttributesFromDB(final JdbcDatabase database) {
+    try (final Stream<MssqlDebeziumStateAttributes> stream = database.unsafeResultSetQuery(
+        connection -> connection.createStatement().executeQuery("select sys.fn_cdc_get_max_lsn()"),
```

Comment:
Should we include a `WAITFOR` here just like in `MssqlCdcTargetPosition.getTargetPosition`?

Reply:
@xiaohansong